bundles/binding/org.openhab.binding.serial/src/main/java/org/openhab/binding/serial/internal/SerialBinding.java

```diff
@@ -145,25 +152,30 @@ public void validateItemType(Item item, String bindingConfig) throws BindingConf
     public void processBindingConfiguration(String context, Item item, String bindingConfig)
             throws BindingConfigParseException {
 
-        int indexOf = bindingConfig.indexOf(',');
-        String serialPart = bindingConfig;
         String pattern = null;
         boolean base64 = false;
-
-        if (indexOf != -1) {
-            String substring = bindingConfig.substring(indexOf + 1);
-            serialPart = bindingConfig.substring(0, indexOf);
-
-            if (substring.startsWith("REGEX(")) {
-                pattern = substring.substring(6, substring.length() - 1);
-            }
-
-            if (substring.equals("BASE64")) {
-                base64 = true;
+        String onCommand = null;
+        String offCommand = null;
+
+        String[] split = bindingConfig.split(",");
+        if (split.length > 0) {
+            for (int i = 1; i < split.length; i++) {
+                String part = split[i];
+                String substring = part.substring(0, part.length());
+
+                if (substring.startsWith("REGEX(")) {
+                    pattern = substring.substring(6, substring.length() - 1);
+                } else if (substring.equals("BASE64")) {
+                    base64 = true;
+                } else if (substring.startsWith("ON(")) {
```

Comment:
These substring matches are error-prone. You should consider beefing up the parsing.

However, I recognize you've just expanded on what already existed here, so this is not required.